src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1242,12 +1249,23 @@ public void RejectRequest(RequestRejectionReason reason, string value)
             throw ex;
         }
 
+        public void SetBadRequestState(RequestRejectionReason reason)
+        {
+            SetBadRequestState(BadHttpRequestException.GetException(reason));
+        }
+
         public void SetBadRequestState(BadHttpRequestException ex)
         {
-            StatusCode = ex.StatusCode;
+            // Setting status code will throw if response has already started
```

Comment:
Yikes. Good thing you caught this now. That could affect bad request bodies too.